### ISSUE TITLE
Does mg work for the diffusion of predicting noise such as stable diffusion?

### ISSUE THREAD

**Issue by hdjsjyl** [tzco/Diffusion-wo-CFG#8]
Hi Author,
    Thanks for your excellent work. 
    Does mg work for the diffusion of predicting noise such as stable diffusion?
    I think your work is very useful. So I tried it in finetuning stable diffusion, but from the first epoch, the result is like this:

![Image](https://github.com/user-attachments/assets/e222e07b-0713-4469-b037-a6af808ab9f1)

I used w=1.5, without dropping label, for each iteration, I will apply target_noise += w*(cond_pred - uncond_pred) as supervision signal.
Do you have any suggestions for the parameter and detail implementation?
Since the pretrained sd model has the target of predicting noise, now while finetuning, I change the target_noise with a regularization item which doesn't produce gradients, do you think it works? 

Best

**Comment by tzco** [comment]
Thanks for your question. It should also work for noise-prediction diffusion models, as we have trained DiT models and find it useful.

I think you can debug from smaller values for `w` starting from 0, since `w=0` corresponds to vanilla training and should be definitely no problem, then incrementing by 0.1 or 0.25. Besides, I notice that you did not drop labels. I am not so sure about finetuning SD models, but our best DiT models trained from scratch should also drop labels. May I ask which codebase and SD model you are using if it is OK?
